core/tests/protractor_utils/waitFor.js

```diff
@@ -123,12 +124,16 @@ var elementAttributeToBe = async function(
 * Wait for new tab is opened
 */
 var newTabToBeCreated = async function(errorMessage, urlToMatch) {
+  var handles = await browser.driver.getAllWindowHandles();
+  await browser.waitForAngularEnabled(false);
+  await browser.switchTo().window(await handles.pop());
+  // This will wait for the page to load completely and then it will fetch
+  // the url.
+  /* eslint-disable-next-line oppia/protractor-practices */
+  await browser.sleep(10000);
+  await browser.waitForAngularEnabled(true);
   await browser.wait(async function() {
-    var handles = await browser.driver.getAllWindowHandles();
-    await browser.waitForAngularEnabled(false);
-    await browser.switchTo().window(await handles.pop());
     var url = await browser.getCurrentUrl();
-    await browser.waitForAngularEnabled(true);
     return await url.match(urlToMatch);
   }, DEFAULT_WAIT_TIME_MSECS, errorMessage);
 };
```

Comment:
The flake was because the url was being fetched before the page loads completely. Hence, waited for the page to oad completely and then fetched the url above